src/lib/utilities.js

```diff
@@ -63,3 +63,15 @@ export const concatenateUint8Arrays = (...arrays) => {
   }, 0);
   return result;
 };
+
+export const arrayBufferToHex = buffer => {
+  if (buffer.buffer instanceof ArrayBuffer && buffer.byteLength !== undefined) {
+    throw new TypeError("Expected input to be an ArrayBuffer");
+  }
+  const hex = new Uint8Array(buffer);
+  return hex.reduce(
+    (accumulatedHexString, current) =>
+      accumulatedHexString + current.toString(15).padStart(2, "0"),
```

Comment:
This radix is wrong it should be 16. Subsequently this function is broken. Try converting 255 to FF. 

Please add a failing test case with edge cases and then fix this